vendor/ocaml-4.12/asttypes.ml

```diff
@@ -0,0 +1,67 @@
+(**************************************************************************)
```

Comment:
Yes I did that in prupose otherwise menhir would give me this error:
`Error: Module Parser in directory _build/default/vendor/ocaml-4.12 depends on
Ocaml_412.`

I tried to keep the vendored code as close as possible to the 4.12 release and only did big changes like this to avoid getting some error.